R/get-majority-t1d-diagnoses.R

```diff
@@ -0,0 +1,43 @@
+#' Determine Majority of Type 1 Diabetes Diagnoses
+#'
+#' This function evaluates whether an individual has a majority of type 1 diabetes-specific hospital diagnoses (DE10) among all type-specific diabetes primary diagnoses (DE10 & DE11) from medical departments.
+#' If an individual has any type-specific diabetes diagnoses from endocrinology departments, the majority is determined only among these contacts.
+#'
+#' @param n_t1d_endocrinology Numeric vector. The count of type 1 diabetes primary diagnoses from endocrinology departments.
+#' @param n_t2d_endocrinology Numeric vector. The count of type 2 diabetes primary diagnoses from endocrinology departments.
+#' @param n_t1d_medical Numeric vector. The count of type 1 diabetes primary diagnoses from other medical departments.
+#' @param n_t2d_medical Numeric vector. The count of type 2 diabetes primary diagnoses from other medical departments.
+#'
+#' @return A logical vector indicating whether the majority of primary diagnoses are related to type 1 diabetes.
+#' @keywords internal
+#'
+#' @examples
+#' # Example usage within the include_diabetes_diagnoses() pipeline:
+#' rbind(lpr2_diabetes_processed, lpr3_diabetes_processed) |>
+#' count_primary_diagnoses_by_department() |>
+#'  dplyr::mutate(
+#'    majority_t1d_primary_diagnoses = get_majority_of_t1d_diagnoses(
+#'      n_t1d_endocrinology,
+#'      n_t2d_endocrinology,
+#'      n_t1d_medical,
+#'      n_t2d_medical
+#'    )
+#'  ) |>
+#'  dplyr::arrange(.data$pnr)
+#'
+#' @seealso [dplyr::case_when()] for condition handling.
+get_majority_of_t1d_diagnoses <- function(n_t1d_endocrinology,
```

Comment:
I thought this was happening after all the inclusions, based on the flow diagram and the doc you wrote out in the function flow. So I'm a bit confused.